src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -36696,6 +36762,30 @@ static void F(Enumerable e)
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void ForEach_UnconstrainedTypeParameter()
+        {
+            var source =
+@"class C<T>
+{
+    void M(T parameter)
+    {
+        foreach (T local in new[] { parameter })
+        {
+        }
+
+        if (parameter == null) throw null;
+        foreach (T local2 in new[] { parameter })
+        {
+        }
+    }
+}";
+            // https://github.com/dotnet/roslyn/issues/29850 Expecting a W warning for assigning possibly null value to local
```

Comment:
>// https://github.com/dotnet/roslyn/issues/29850 Expecting a W warning for assigning possibly null value to local [](start = 12, length = 113)

This shouldn't be a warning since `local` has the same nullability as `parameter`.